### PR TITLE
`General`: Improve the appearance of system notifications

### PR DESCRIPTION
**Pull request by bassner**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

System notifications are shown on top of the page on all clients, e.g. to inform about an upcoming maintenance period.
Currently, they look like our alerts looked previously, and they are not working properly at the moment:
- Only one notification is supported at the same time
- Updating the shown notifications via websocket was not implemented properly, and often a reload of the app was required to make it show up.
- The management interface nor the server side endpoints did do any proper user input validation

grafik

### Description

I've decided to give it a full overhaul.
The following changes have been added:

- Changed appearance of the notifications: They are now glued to the bottom of the navbar and look more modern; they also got a intro animation as moving things usually raise a users attention more effectively that a colorful, but static item
- Notifications are now temporarily dismissable, but will show up again on a reload or system notification update
- Notifications with a date in the future will now show up exactly in time without a reload
- The management view for system notifications has been updated:
  * Added three notification states: Scheduled, active, and expired (previously: Active, Not Active)
  * The list of notifications is now sorted by notification date by default
  * The update component now validates that the type is not empty, and that the expiration date is after the notification date

### Steps for Testing

Prerequisites:
- 1 Admin

1. Log in to Artemis
2. Navigate to Server Administration > System notification
3. Create multiple new system notifications. Play around with the types, the title, the text (which is optiona), and the dates: Make them active right after creating them; make one active a minute in the future for example and check that it shows up in time; make one that expires in a minute and check that it actually disappears in time ....... etc)
4. Make sure that you can not set an expiration date that is before the notification date. You can even try to modify the request, the server should check it as well
5. Verify that the "Active", "Scheduled" and "Expirted" batches are shown correctly and update live as well
6. Verify that you can temporatily dismiss a notification and it shows up again after a reload
7. Verify that all dismissed notifications show up again when you update a notification
8. Log out, check that the notifications are visible if you're not logged in as well (after a fresh reload)
9. Check that the notifications wrap correctly on mobile phones: The title and the icon should move into the first line, the text below it

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| system-notification.component.ts | 70.83 | 94.82 |
| system-notification-management-update.component.ts | 86.25% | 88.88% |
| system-notification-management.component.ts | 70.37% | 84.81% |

### Screenshots

Screenshot 2022-07-22 at 18 48 01
Screenshot 2022-07-22 at 17 27 05

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/SystemNotificationIntegrationTest.java [hunk 1]
```diff
@@ -71,12 +73,15 @@ void testGetActiveSystemNotification() throws Exception {
...
         // Do the actual request that is tested here.
-        SystemNotification notification = request.get("/api/system-notifications/active-notification", HttpStatus.OK, SystemNotification.class);
+        List<SystemNotification> notification = request.getList("/api/system-notifications/active", HttpStatus.OK, SystemNotification.class);
 
         // The returned notification must be an active notification.
-        assertThat(systemNotification.getExpireDate()).as("Returned notification has not expired yet.").isAfterOrEqualTo(ZonedDateTime.now());
-        assertThat(systemNotification.getNotificationDate()).as("Returned notification is active.").isBeforeOrEqualTo(ZonedDateTime.now());
-        assertThat(notification).as("Returned notification is active system notification.").isEqualTo(systemNotificationActive);
+        assertThat(notification).hasSize(2);
+        assertThat(notification).as("Returned notification is active system notification.").isEqualTo(List.of(systemNotificationActive, systemNotificationFuture));
```
Comment (Strohgelaender) [inline src/test/java/de/tum/in/www1/artemis/SystemNotificationIntegrationTest.java:80]:
```suggestion
        assertThat(notification).hasSize(2)
            .as("Returned notification is active system notification.").containsExactly(systemNotificationActive, systemNotificationFuture);
```

Comment (bassner) [reply src/test/java/de/tum/in/www1/artemis/SystemNotificationIntegrationTest.java]: Done! :)

**Review by JohannesStoehr — APPROVED** [review]
Code changes look good

**Review by joaquinmurcia — APPROVED** [review]
Tested on TS2. Works as expected.

**Pull request merged**